frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -577,7 +582,47 @@ export default class IndexController extends Controller {
     this.journalsContainerTarget.classList.add('work-packages-activities-tab-index-component--journals-container_with-input-compensation');
   }
 
-  setLastUpdateTimestamp() {
+  private setLastUpdateTimestamp() {
     this.lastUpdateTimestamp = new Date().toISOString();
   }
+
+  // Towards the code below:
+  // Ideally the stem rendering would be correctly rendered for all UI states from the server
+  // but as we push single elements into the DOM via turbo-streams, the server-side rendered collection state gets stale quickly
+  // I've decided to go with a client-side rendering-correction approach for now
+  // as I don't want to introduce more complexity and queries (n+1 for position checks etc.) to the server-side rendering
```

Comment:
I reckon this is fine for a first fix- but in the end, server rendering would be more stable and less complex IMO. Right now this controller is quite complex from my POV; which will hurt maintenance.

Let's file a code maintenance ticket and tag it to the next version- for simplifying this.